light_sdk/src/reading/mod.rs

```diff
@@ -0,0 +1,679 @@
+                .map_err(|e| Error::Other(e.to_string()))?,
+        )
+        .map_err(|e| Error::Other(e.to_string()))?;
+        let rt = Runtime::new().unwrap();
+        rt.block_on(rpc::get_delegators_delegation_at(&client, address, epoch))
+    }
+
+    /// Query and return validator's commission rate and max commission rate
+    /// change per epoch
+    pub fn query_commission_rate(
+        tendermint_addr: &str,
+        validator: &Address,
+        epoch: Option<Epoch>,
+    ) -> Result<Option<CommissionPair>, Error> {
+        let client = HttpClient::new(
+            TendermintAddress::from_str(tendermint_addr)
+                .map_err(|e| Error::Other(e.to_string()))?,
+        )
+        .map_err(|e| Error::Other(e.to_string()))?;
+        let rt = Runtime::new().unwrap();
+        rt.block_on(rpc::query_commission_rate(&client, validator, epoch))
+    }
+
+    /// Query and return validator's metadata, including the commission rate and
+    /// max commission rate change
+    pub fn query_metadata(
+        tendermint_addr: &str,
+        validator: &Address,
+        epoch: Option<Epoch>,
+    ) -> Result<(Option<ValidatorMetaData>, Option<CommissionPair>), Error>
+    {
+        let client = HttpClient::new(
+            TendermintAddress::from_str(tendermint_addr)
+                .map_err(|e| Error::Other(e.to_string()))?,
+        )
+        .map_err(|e| Error::Other(e.to_string()))?;
+        let rt = Runtime::new().unwrap();
+        rt.block_on(rpc::query_metadata(&client, validator, epoch))
+    }
+
+    /// Query and return the incoming redelegation epoch for a given pair of
+    /// source validator and delegator, if there is any.
+    pub fn query_incoming_redelegations(
+        tendermint_addr: &str,
+        src_validator: &Address,
+        delegator: &Address,
+    ) -> Result<Option<Epoch>, Error> {
+        let client = HttpClient::new(
+            TendermintAddress::from_str(tendermint_addr)
+                .map_err(|e| Error::Other(e.to_string()))?,
+        )
+        .map_err(|e| Error::Other(e.to_string()))?;
+        let rt = Runtime::new().unwrap();
+        rt.block_on(rpc::query_incoming_redelegations(
+            &client,
+            src_validator,
+            delegator,
+        ))
+    }
+
+    /// Query a validator's bonds for a given epoch
+    pub fn query_bond(
+        tendermint_addr: &str,
+        source: &Address,
+        validator: &Address,
+        epoch: Option<Epoch>,
+    ) -> Result<token::Amount, Error> {
+        let client = HttpClient::new(
+            TendermintAddress::from_str(tendermint_addr)
+                .map_err(|e| Error::Other(e.to_string()))?,
+        )
+        .map_err(|e| Error::Other(e.to_string()))?;
+        let rt = Runtime::new().unwrap();
+        rt.block_on(rpc::query_bond(&client, source, validator, epoch))
+    }
+
+    /// Query a validator's unbonds for a given epoch
+    pub fn query_and_print_unbonds(
```

Comment:
let's not have print options in the api. if a caller wants to print something, they can do it in their own code.